bitcoin-magazine/bitcoin-explained/bitcoin-core-24.0.md

```diff
@@ -0,0 +1,1150 @@
+
+Aaron van Wirdum: 00:05:29
+
+Yeah, I'm just saying we made a selection of some of the changes that we think are probably the most interesting
+to you, our dear listener.
+
+## Peer-to-Peer network changes
+
+Sjors Provoost: 00:05:38
+
+That's right.
+
+Aaron van Wirdum: 00:05:40
+
+Okay, so let's start.
+Sjors, I think that this one is the first one mentioned in the release notes as well.
+It's about the peer-to-peer and network changes.
+Is that a typo?
+Should you just say peer-to-peer changes?
+Peer-to-peer network changes?
+Whatever.
+
+Sjors Provoost: 00:05:55
+
+Peer-to-peer networking.
+
+Aaron van Wirdum: 00:05:57
+
+Let's move on.
+So it says that there's a change and you're gonna explain to me what this is That's the format of the podcast in case you forgot.
+It says download headers from peers so the downloading from Downloading of headers from peers has been reworked.
+That's right somehow.
+
+## Download headers from peers
+
+Sjors Provoost: 00:06:18
+
+Yeah, this one might be worth a separate episode, but I think we can explain the gist of it now.
+So when your node first starts up, it is going to find peers, which we discussed in an earlier episode, and then it's going to download headers from peers, which we've also discussed in earlier episodes.
+So it tries to get all the headers first.
+
+Aaron van Wirdum: 00:06:36
+
+Hang on, so every block has a header which is essentially the hash of the block itself, right?
+
+Sjors Provoost: 00:06:41
+
+Plus a little bit of extra info that, for example, contains the timestamp and the amount of proof of work.
+Well, the amount of proof of work follows in the hash.
+
+Aaron van Wirdum: 00:06:52
+
+It's basically hash plus some extra info.
+Yes.
+So it's not the whole block.
+Okay.
+So first you start downloading not the whole block, or not all of the blocks, but you start downloading the headers, right?
+
+Sjors Provoost: 00:07:04
+
+Yes, and the idea is that you first want to make sure that the chain that you're following has enough proof of work that it's even worth downloading the blocks for.
+Now that's been improved many years ago.
+It used to be that I think it would just download headers and then some blocks and then some headers and then some blocks and that has been changed many years ago to first download all the headers and only then when you know for sure that you have enough proof of work because you can check that based on the headers then you start downloading the blocks.
+This prevents you from wasting time chasing and basically chasing dead ends.
+
+Aaron van Wirdum: 00:07:32
+
+Okay, that was already the case.
+So even in Bitcoin Core 23, you would already start with downloading only the headers and only later the blocks.
+Okay, so what has changed?
+
+## Download headers twice
+
+Sjors Provoost: 00:07:41
+
+So what has changed now is that we're going to download the headers twice.
```

Comment:
The word 'So' is frequently used as a filler at the beginning of sentences, often without adding meaningful content. While it occasionally serves a purpose, in many cases, removing it enhances readability.